.github/workflows/windows-msvc.yml

```diff
@@ -14,81 +14,106 @@ jobs:
+          key: ${{ runner.os }}-gtk
+          restore-keys: |
+            ${{ runner.os }}-gtk
 
+      - name: Set up the PATH environment
+        run: |
+          echo "C:\pkg-config-lite-0.28-1\bin" >> $GITHUB_PATH
+          echo "C:\gnome\bin" >> $GITHUB_PATH
+        shell: bash
+
+      - name: Install pkgconfig-lite
+        run: |
+          Invoke-WebRequest -Uri https://deac-fra.dl.sourceforge.net/project/pkgconfiglite/0.28-1/pkg-config-lite-0.28-1_bin-win32.zip -OutFile /pkg_config_lite.zip
+          Expand-Archive /pkg_config_lite.zip -DestinationPath C:\
+          ls C:\
+          ls C:\pkg-config-lite-0.28-1
+          ls C:\pkg-config-lite-0.28-1\bin
+          pkg-config --version
+
+      - name: Clone GTK
+        working-directory: /
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: |
+          git clone -b 4.6.0 https://gitlab.gnome.org/GNOME/gtk.git --depth 1
+
+      - name: Clone libadwaita
+        if: steps.cache.outputs.cache-hit != 'true'
+        working-directory: /
+        run: |
+          git clone -b 1.1.0 https://gitlab.gnome.org/GNOME/libadwaita.git --depth 1
+
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.x"
+
+      - name: Install Python Dependencies
+        run: pip install meson ninja
+
+      - name: Setup MSVC
+        uses: bus1/cabuild/action/msdevshell@v1
+        with:
+          architecture: x64
+
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          components: clippy
+
+      - name: Prepare GTK build
+        working-directory: /gtk
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: |
+          meson setup builddir --prefix=C:/gnome -Dbuild-tests=false -Dmedia-gstreamer=disabled -Dbuild-examples=false
+
+      - name: Build and install GTK
+        working-directory: /gtk
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: |
+          meson install -C builddir
+
+      - name: Prepare libadwaita build
+        if: steps.cache.outputs.cache-hit != 'true'
+        working-directory: /libadwaita
+        run: |
+          meson setup builddir --prefix=C:/gnome -Dintrospection=disabled -Dvapi=false -Dexamples=false -Dtests=false
+
+      - name: Build and install libadwaita
+        if: steps.cache.outputs.cache-hit != 'true'
+        working-directory: /libadwaita
+        run: |
+          meson install -C builddir
+
+      - name: Build (workspace)
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --features v4_6
+      - name: Clippy (workspace)
+        uses: actions-rs/cargo@v1
+        with:
+          command: clippy
+          args: --features v4_6
+      - name: Tests (workspace)
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --features v4_6
+
+      - name: "Build (book)"
```

Comment:
Does it matter to build the book under windows?

Reply:
Mmmh fair point, we probably only verify that way that libadwaita continues to build on Windows

Reply:
That is something up to libadwaita/libadwaita-rs to fix

Reply:
I've removed it